### DISCUSSION TITLE
Bug running a single phase in temperature range

### DISCUSSION THREAD

**Discussion by gplanu** [pycalphad/pycalphad#439] [Q&A]
Dear PyCalphad Community,
I have a new issue to show you when I try to calculate the equilibrium of a complex steel composition mixture along a temperature array.
I share with you below the code I prepared to make these simple operations, I do not report how to read a database (.tdb) in pycalphad, but only y starting inputs.
1. Compositions
![image](https://user-images.githubusercontent.com/117755633/204015156-bf7f8da3-4fec-4d55-919f-67e961731344.png)
2. Read composition from elComp file and conversion in mole fractions
![image](https://user-images.githubusercontent.com/117755633/204015448-156c119d-bcb2-457e-9c4e-c52b9305c1d2.png)
3. Components definition
![image](https://user-images.githubusercontent.com/117755633/204015801-594e49ee-3d9a-4917-96f7-b2dce1a1a2f7.png)
4. Conditions: consider v.T: (300, 2000, 50)
![image](https://user-images.githubusercontent.com/117755633/204020990-9660aefa-67be-4c2a-b579-98993bbcdbd0.png)

5. Exception during evaluation:
![image](https://user-images.githubusercontent.com/117755633/204021097-7f40b6e1-a482-446c-9141-3e7d31516aed.png)

Working with single temperature it works with output some NaN values and it works also with FCC_A1 phase.
Thanks in advance for your help, 
Looking forward for your suggestions,
Best regards.
Giulio

**Comment by richardotis** [comment]
Can you share the database you're using to do these calculations?

**Reply by gplanu** [reply]
Yes sure, the tdb is this one -> https://www.matcalc.at/images/stories/Download/Database/mc_fe_v2.059.tdb
Looking forward your suggestions.
Thanks in advance.

**Reply by richardotis** [reply]
Could you try this version of the database to see if you get any differences? https://github.com/materialsgenomefoundation/2021-workshop-material/blob/98f4b8adf2526b996aee5a3fd25ed905a4bf5127/pycalphad/databases/mc_fe_v2.059.pycalphad.tdb

**Reply by gplanu** [reply]
Hello Richard, ok I can try this afternoon or tomorrow morning. Which are the main differences?

**Reply by gplanu** [reply]
Dear Richard, I tried with your db the same conditions. Unfortunately it happens again:
![image](https://user-images.githubusercontent.com/117755633/204344383-ec3d1c93-dd2b-450f-880f-b08b313069ea.png)

**Reply by gplanu** [reply]
Dear Richard, any news regarding this topic?

**Reply by richardotis** [reply]
I recommend waiting for this pull request to be merged: https://github.com/pycalphad/pycalphad/pull/432
There will be some fixes to how single-phase energies are computed that may affect this issue.

**Reply by gplanu** [reply]
Thanks Richard, do you have in mind when it could be available?

**Reply by richardotis** [reply]
I expect some progress in the next few weeks.

**Reply by gplanu** [reply]
Dear Richard, thanks for the answer, please let me know when it will be ready, secondly regarding the starting problem about both the equilibrium convergence method i tried also to modify the grid_opt parameter pden from 60 to 120 as suggested by you, but it does not solve the issue.
